api/src/client/dynamics/license-status/RgbDynamicsLicenseApplicationIdsClient.test.ts

```diff
@@ -0,0 +1,449 @@
+        },
+        {
+          expirationDateISO: application_2.rgb_expirationdate,
+          applicationId: application_2.rgb_applicationid,
+          licenseStatus: "ACTIVE",
+          professionNameAndLicenseType: "Health Club Services",
+        },
+      ]);
+      expect(mockAxios.get).toHaveBeenCalledTimes(1);
+    });
+
+    it("makes multiple get requests and returns all matching license application ids", async () => {
+      const application_1 = generateLicenseApplicationIdApiResponseValue({
+        rgb_name: "Example Business-Public Movers",
+        rgb_number: "123400",
+        statuscode: 100000006,
+        _rgb_apptypeid_value: HEALTH_CLUB_LICENSE_UUID,
+      });
+
+      const application_2 = generateLicenseApplicationIdApiResponseValue({
+        rgb_name: "Example Business-Cosmetology",
+        rgb_number: "123400",
+        statuscode: 100000006,
+        _rgb_apptypeid_value: HEALTH_CLUB_LICENSE_UUID,
+      });
+
+      mockAxios.get
+        .mockResolvedValueOnce({
+          data: {
+            value: [application_1],
+          },
+        })
+        .mockResolvedValueOnce({
+          data: {
+            value: [application_2],
+          },
+        });
+
+      expect(
+        await client.getLicenseApplicationIdsForAllBusinessIds(mockAccessToken, [
+          mockBusinessNameAndId_1,
+          mockBusinessNameAndId_2,
+        ])
+      ).toEqual([
+        {
+          applicationId: application_1.rgb_applicationid,
+          expirationDateISO: application_1.rgb_expirationdate,
+          licenseStatus: "ACTIVE",
+          professionNameAndLicenseType: "Health Club Services",
+        },
+        {
+          applicationId: application_2.rgb_applicationid,
+          expirationDateISO: application_2.rgb_expirationdate,
+          licenseStatus: "ACTIVE",
+          professionNameAndLicenseType: "Health Club Services",
+        },
+      ]);
+      expect(mockAxios.get).toHaveBeenCalledTimes(2);
+    });
+
+    it("throws status error when api call fails", async () => {
+      mockAxios.get.mockRejectedValue({
+        response: { status: 500 },
+      });
+
+      await expect(
+        client.getLicenseApplicationIdsForAllBusinessIds(mockAccessToken, [mockBusinessNameAndId_1])
+      ).rejects.toEqual(500);
+    });
+  });
+
+  describe("licenseAppType", () => {
+    it("validates LICENSE_APPLICATION_INFORMATION contains all supported licenses", () => {
+      expect(RGB_LICENSE_APPLICATION_INFORMATION.length).toBe(Object.keys(licenseAppType).length);
+      for (const application of RGB_LICENSE_APPLICATION_INFORMATION) {
+        const licenseAppTypeKey = application.appTypeCode
+          ? `${application.uuid}-${application.appTypeCode}`
+          : application.uuid;
+        expect(licenseAppType[licenseAppTypeKey]).toBe(application.expectedProfession);
+      }
+    });
+
+    for (const application of RGB_LICENSE_APPLICATION_INFORMATION) {
```

Comment:
dust - probably better as a it.each?

Reply:
We initially did this because I didn't know you could actually access object props in the test name when using it.each(). 

Leaving a note to share, but also help myself remember.

```typescript
export const RGB_LICENSE_APPLICATION_INFORMATION = [
  {
    uuid: BILL_LICENSE_UUID,
    appTypeCode: null,
    expectedProfession: "Bill",
  }
]

it.each(RGB_LICENSE_APPLICATION_INFORMATION)(
      "returns object whose professionNameAndLicenseType key is $expectedProfession",
      async ({ appTypeCode, expectedProfession, uuid }) => {
     // test logic goes here
};
```